src/swoop/api/io.py

```diff
@@ -0,0 +1,107 @@
+import io
+import logging
+
+from minio import Minio
+from minio.error import S3Error
+
+logger = logging.getLogger("uvicorn")
+logger.setLevel(logging.INFO)
+
+
+class IOClient:
+    client = None
+    bucket_name = None
+
+    def __init__(
+        self, s3_endpoint: str, access_key: str, secret_key: str, bucket_name=None
+    ):
+        """Initialize IO Client."""
+        self.client = Minio(
+            s3_endpoint.split("//").pop() if "http" in s3_endpoint else s3_endpoint,
+            access_key=access_key,
+            secret_key=secret_key,
+            secure="http://" not in s3_endpoint,
+        )
+        self.bucket_name = bucket_name
+
+    def get_object(self, object_name: str):
+        """Retrieve from object storage."""
+        object_response = None
+        if (
+            object_name
+            and self.bucket_name
+            and self.client.bucket_exists(self.bucket_name)
+        ):
+            try:
+                response = self.client.get_object(self.bucket_name, object_name)
+                object_response = response.json()
+                logger.debug(f"retrieved object content: {object_response}")
+            except (S3Error, UnboundLocalError) as err:
+                logger.debug(err)
+            else:
+                response.close()
+                response.release_conn()
```

Comment:
Per the idea to make the bucket name required and ensure the bucket exists:

```suggestion
        if not object_name:
            raise ValueError("object_name cannot be empty"

        try:
            response = self.client.get_object(self.bucket_name, object_name)
            object_response = response.json()
            logger.debug(f"retrieved object content: {object_response}")
        except (S3Error, UnboundLocalError) as err:
            logger.debug(err)
        finally:
            response.close()
            response.release_conn()
        else:
            object_response = response.json()
            logger.debug("retrieved object content: %s", object_response)
```

Also, I think my suggestion around the `else` was misunderstood, so I tweaked the above for that as well.

Lastly, I noticed the use of the f-string in the log message. To avoid the overhead of templating the message string for a log function that doesn't get called (when the logging level is > debug), use the logging string templating support for lazy evaluation. See my change to the logger call as an example.